src/Core/Domain/Customer/ValueObject/Email.php

```diff
@@ -26,97 +26,16 @@
 
 namespace PrestaShop\PrestaShop\Core\Domain\Customer\ValueObject;
 
-use PrestaShop\PrestaShop\Core\Domain\Customer\Exception\CustomerConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\ValueObject\Email as BaseEmail;
 
 /**
- * Stores customer's email address
+ * @deprecated since version 1.7.7 use PrestaShop\PrestaShop\Core\Domain\ValueObject\Email instead
  */
-class Email
+class Email extends BaseEmail
```

Comment:
Since this class is now deprecated, it would be better if we stop using it in the core code, I only found a reference in `GetCustomerForAddressCreation` but maybe there are more

Reply:
@jolelievre Yeah, I didn't find this class elsewhere, and even in `GetCustomerForAddressCreation`, it's not really used because phpdoc of the function `getCustomerEmail()` is says that it returns a `Email` but the function declares returning a `string`. I'll update the phpddoc!